debug/debugger.go

```diff
@@ -0,0 +1,815 @@
+// Copyright 2024 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package debug
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/open-policy-agent/opa/ast/location"
+	fileurl "github.com/open-policy-agent/opa/internal/file/url"
+	"github.com/open-policy-agent/opa/logging"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/topdown"
+	prnt "github.com/open-policy-agent/opa/topdown/print"
+)
+
+type Debugger interface {
```

Comment:
For Session, Threads etc. some linking to the spec: https://microsoft.github.io/debug-adapter-protocol/specification would be helpful for future readers I'd imagine. Just so it's clear that this implementation is following that spec rather than something OPA-specific.